### PR TITLE
new helper module, full avro 1.4-1.9 support

### PR DESCRIPTION
**Pull request by radai-rosenblatt**
this commit includes the following:
- new helper module build has a sub-module per major version of avro which
  compiles natively vs the target avro version. this eliminates the need
  for reflection in _most_ use cases. cases involving optional avro-compiler
  jar still rely on reflection
- helper code now implemented across all major avro versions from 1.4 to
  1.9 inclusive
- code generation support has been extended to cover above avro versions
- tests are now performed with multiple classpaths, including variants
  with and without the avro-compiler jar (on avro 1.5+)

Signed-off-by: Radai Rosenblatt <rrosenbl@rrosenbl-ld1.linkedin.biz>

### CODE REVIEW

**Comment by radai-rosenblatt** [comment]
loose ends still missing:

- re-enable the rest of the modules :-)
- fat-jar helper into a single module
- build cleanups

File: helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/TemplateUtil.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2018 LinkedIn Corp.
```
Comment (smccauliff) [inline helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/TemplateUtil.java]: 2020?

**Comment by smccauliff** [comment]
This needs more code.

File: helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/AvroGeneratedSourceCode.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2018 LinkedIn Corp.
```
Comment (akaritakai) [inline helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/AvroGeneratedSourceCode.java]: 2020?

**Comment by radai-rosenblatt** [comment]
@akaritakai - tests include java classes generated by avro X, made compatible, and run under avro Y for all X and Y. and change to code generation would cause a compilation/runtime failure

**Pull request merged**